dhcpv4/dhcpv4_test.go

```diff
@@ -322,6 +322,30 @@ func TestGetOption(t *testing.T) {
 	require.Equal(t, d.GetOneOption(OptionRouter), nil)
 }
 
+func TestDHCPv4RequestFromOffer(t *testing.T) {
+	offer, err := New()
+	require.NoError(t, err)
+	offer.AddOption(&OptMessageType{MessageType: MessageTypeOffer})
+	offer.AddOption(&OptServerIdentifier{ServerID: net.IPv4(192, 168, 0, 1)})
+	req, err := RequestFromOffer(*offer)
+	require.NoError(t, err)
+	require.NotEqual(t, (*MessageType)(nil), *req.MessageType())
```

Comment:
Why the explicit cast on the `nil` here? If it's a pointer can't it just be compared to nil?